qiskit_nature/circuit/library/initial_states/hartree_fock.py

```diff
@@ -26,49 +29,47 @@ class HartreeFock(QuantumCircuit):
     """A Hartree-Fock initial state."""
 
     def __init__(self,
-                 num_orbitals: int,
+                 num_spin_orbitals: int,
                  num_particles: Union[Tuple[int, int], int],
-                 qubit_mapping: str = 'parity',
-                 two_qubit_reduction: bool = True,
-                 sq_list: Optional[List[int]] = None) -> None:
+                 qubit_converter: QubitConverter) -> None:
         """
         Args:
-            num_orbitals: The number of spin orbitals, has a min. value of 1.
+            num_spin_orbitals: The number of spin orbitals, has a min. value of 1.
             num_particles: The number of particles. If this is an integer, it is the total (even)
                 number of particles. If a tuple, the first number is alpha and the second number is
                 beta.
-            qubit_mapping: The mapping type for qubit operator.
-            two_qubit_reduction: A flag indicating whether or not two qubit is reduced.
-            sq_list: The position of the single-qubit operators that anticommute with the cliffords.
+            qubit_converter: a QubitConverter instance.
         """
+
         # get the bitstring encoding the Hartree Fock state
-        bitstr = hartree_fock_bitstring(num_orbitals, num_particles, qubit_mapping,
-                                        two_qubit_reduction, sq_list)
+        bitstr = hartree_fock_bitstring(num_spin_orbitals, num_particles)
+
+        # encode the bitstring as a `FermionicOp`
+        label = ['+' if bit else 'I' for bit in bitstr]
+        bitstr_op = SecondQuantizedOp([FermionicOp(''.join(label))])
+
+        # map the `FermionicOp` to a qubit operator
+        qubit_op = qubit_converter.to_qubit_ops([bitstr_op])[0]
 
         # construct the circuit
-        qr = QuantumRegister(len(bitstr), 'q')
+        qr = QuantumRegister(qubit_op.num_qubits, 'q')
         super().__init__(qr, name='HF')
 
-        # add gates in the right positions
-        for i, bit in enumerate(reversed(bitstr)):
+        # Add gates in the right positions: we are only interested in the `X` gates because we want
+        # to create particles (0 -> 1) where the initial state introduced a creation (`+`) operator.
+        for i, bit in enumerate(qubit_op.primitive.table.X[0]):
             if bit:
                 self.x(i)
 
 
-def hartree_fock_bitstring(num_orbitals: int,
-                           num_particles: Union[Tuple[int, int], int],
-                           qubit_mapping: str = 'parity',
-                           two_qubit_reduction: bool = True,
-                           sq_list: Optional[List[int]] = None) -> List[bool]:
+def hartree_fock_bitstring(num_spin_orbitals: int,
+                           num_particles: Union[Tuple[int, int], int]) -> List[bool]:
```

Comment:
When we first started chemistry, and only did closed shell, we just had the int, but when we added open shell, and needed different number for alpha and beta, where the tuple was added, it was always our intent to deprecate supporting int and require a tuple. I do not mind whether its kept or not, but the stack always emitted num particles as a tuple which you would pass onto these methods, and hence a user would need to have created an int outside of that.

Reply:
Sure I will update that and deprecate the `int`. I was confused to see it there but forgot to check again later :+1: